File: src/test/java/fr/sogilis/blog/article/kotlinobjectmother/TownMother.java
package fr.sogilis.blog.article.kotlinobjectmother;

public class TownMother {

    public static TownBuilder one() {
        return new TownBuilder()
            .name("Vannes");
    }

    public static TownBuilder kaamelott() {
        return one()
            .name("Kaamelott");
    }
}

File: src/test/kotlin/fr/sogilis/blog/article/kotlinobjectmother/TownMother.kt
package fr.sogilis.blog.article.kotlinobjectmother

object TownMother {
    fun one() = TownBuilder().apply {name = "Vannes"}
    fun kaamelott() = one().apply {name = "Kaamelott"}
}

class TownBuilder : Builder<Town> {
    var name: String? = null
    override fun build() = Town(name!!)
}
